lib/cli/dynamic-list.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const inquirer = require('inquirer');
+const Choices = require('inquirer/lib/objects/choices');
+
+const updateChoices = function(choices) {
+  this.opt.choices = new Choices(choices);
+  this.render();
+};
+
+/**
+ * Prompt for `inquirer` that adds dynamic update functionality to the `list` prompt.
+ *
+ * @class
+ * @public
+ * @extends inquirer.prompt.prompts.list
+ *
+ * @example
+ *
+ * inquirer.registerPrompt('dynamicList', DynamicList);
+ *
+ * inquirer.prompt([{
+ *   message: ''What size do you need?'',
+ *   type: 'dynamicList',
+ *   choices: ['Jumbo', 'Large', 'Standard', 'Medium', 'Small', 'Micro'],
```

Comment:
Can we add spaces between the items and the brackets for consistency with the remaining of the codebase?

```js
[ 'Jumbo', 'Large', 'Standard', 'Medium', 'Small', 'Micro' ]
```